model/map/src/main/java/org/keycloak/models/map/userSession/SessionExpiration.java

```diff
@@ -0,0 +1,153 @@
+/*
+ * Copyright 2021 Red Hat, Inc. and/or its affiliates
+ * and other contributors as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.keycloak.models.map.userSession;
+
+import org.keycloak.common.util.Time;
+import org.keycloak.models.ClientModel;
+import org.keycloak.models.RealmModel;
+import org.keycloak.protocol.oidc.OIDCConfigAttributes;
+
+/**
+ * @author <a href="mailto:mkanis@redhat.com">Martin Kanis</a>
+ */
+public class SessionExpiration {
```

Comment:
This seems similar to what we are doing here: https://github.com/keycloak/keycloak/blob/8255cba930196a53d0274be3a1f22b7e39df9c4f/services/src/main/java/org/keycloak/protocol/oidc/TokenManager.java#L701 should we reuse? Maybe in a follow-up task?

Reply:
yeah it's quite similar but not exactly the same. After some refactoring it could be probably re-used. But I vote to do it as follow up :)

Reply:
Should we create a Jira in the cleanup epic for this?